File: redandsilver/easy/java/LongestCommonPrefix.java
package redandsilver.easy.java;

import java.util.Arrays;

class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String f = strs[0];
        String l = strs[strs.length-1];
        StringBuilder answer = new StringBuilder();
        int n = Math.min(f.length(),l.length());
        for(int i=0; i<n; i++){
            if(f.charAt(i) != l.charAt(i)){
                return answer.toString();
            }
            answer.append(f.charAt(i));
        }
        return answer.toString();
    }
}

File: redandsilver/easy/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string answer = "";
        // sort
        sort(strs.begin(),strs.end());
        int n = strs.size();
        string first = strs[0], last = strs[n-1];
        int len = min(first.size(), last.size());

        for(int i=0; i<len; i++){
            if(first[i] != last[i]){
                return answer;
            }
            answer+=first[i];
        }
        return answer;
    }
};